internal/services/monitor/monitor_smart_detector_alert_rule_resource.go

```diff
@@ -301,11 +306,20 @@ func flattenMonitorSmartDetectorAlertRuleActionGroup(input *smartdetectoralertru
 		CustomWebhookPayload = *input.CustomWebhookPayload
 	}
 
+	var groupIds []string
```

Comment:
This wants an explicit empty value/size set:

```suggestion
	groupIds := make([]string, 0)
```